File: CS/MVCxDashboard_DataLoading/App_Start/DashboardConfig.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardWeb;
using DevExpress.DashboardWeb.Mvc;
using System;
using System.Collections.Generic;
using System.Threading;
using System.Web.Routing;

public class DashboardConfig {
    public static void RegisterService(RouteCollection routes) {
        routes.MapDashboardRoute("dashboardControl", "DefaultDashboard");

        DashboardConfigurator.Default.SetDashboardStorage(new DashboardFileStorage(@"~/App_Data/"));

        var dataSourceStorage = new DataSourceInMemoryStorage();
        DashboardConfigurator.Default.SetDataSourceStorage(dataSourceStorage);
        DashboardObjectDataSource objDataSource = new DashboardObjectDataSource("Object Data Source");
        objDataSource.DataSource = typeof(SalesPersonData);
        dataSourceStorage.RegisterDataSource("objDataSource", objDataSource.SaveToXml());

        DashboardConfigurator.Default.DataLoading += Default_DataLoading;
    }

    private static void Default_DataLoading(object sender, DataLoadingWebEventArgs e) {
        if (e.DataSourceName == "Object Data Source") {
            e.Data = CreateData();
        }
    }

    public static List<SalesPersonData> CreateData() {
        List<SalesPersonData> data = new List<SalesPersonData>();
        string[] salesPersons = { "Andrew Fuller", "Michael Suyama",
                                    "Robert King", "Nancy Davolio",
                                    "Margaret Peacock", "Laura Callahan",
                                    "Steven Buchanan", "Janet Leverling" };

        for (int i = 0; i < 100; i++) {
            SalesPersonData record = new SalesPersonData();
            int seed = (int)DateTime.Now.Ticks & 0x0000FFFF;
            record.SalesPerson = salesPersons[new Random(seed).Next(0, salesPersons.Length)];
            record.Quantity = new Random(seed).Next(0, 100);
            data.Add(record);
            Thread.Sleep(3);
        }
        return data;
    }
}

public class SalesPersonData {
    public string SalesPerson { get; set; }
    public int Quantity { get; set; }
}

File: VB/MVCxDashboard_DataLoading/App_Start/DashboardConfig.vb
Imports System.Threading
Imports System.Web.Routing
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardWeb
Imports DevExpress.DashboardWeb.Mvc

Public Class DashboardConfig
    Public Shared Sub RegisterService(ByVal routes As RouteCollection)
        routes.MapDashboardRoute("dashboardControl", "DefaultDashboard")

        DashboardConfigurator.Default.SetDashboardStorage(New DashboardFileStorage("~/App_Data/"))

        Dim dataSourceStorage = New DataSourceInMemoryStorage()
        DashboardConfigurator.Default.SetDataSourceStorage(dataSourceStorage)
        Dim objDataSource As New DashboardObjectDataSource("Object Data Source")
        objDataSource.DataId = "odsSales"
        objDataSource.DataSource = GetType(SalesPersonData)
        dataSourceStorage.RegisterDataSource("objDataSource", objDataSource.SaveToXml())

        AddHandler DashboardConfigurator.Default.DataLoading, AddressOf Default_DataLoading
    End Sub

    Private Shared Sub Default_DataLoading(ByVal sender As Object, ByVal e As DataLoadingWebEventArgs)
        If e.DataId = "odsSales" Then
            e.Data = CreateData()
        End If
    End Sub

    Public Shared Function CreateData() As List(Of SalesPersonData)
        Dim data As New List(Of SalesPersonData)()
        Dim salesPersons() As String = {"Andrew Fuller", "Michael Suyama",
                                            "Robert King", "Nancy Davolio",
                                            "Margaret Peacock", "Laura Callahan",
                                            "Steven Buchanan", "Janet Leverling"}

        For i As Integer = 0 To 99
            Dim record As New SalesPersonData()
            Dim seed As Long = CLng(Date.Now.Ticks) And &HFFFF
            record.SalesPerson = salesPersons((New Random(CInt(seed))).Next(0, salesPersons.Length))
            record.Quantity = (New Random(CInt(seed))).Next(0, 100)
            data.Add(record)
            Thread.Sleep(3)
        Next i
        Return data
    End Function
End Class

Public Class SalesPersonData
    Public Property SalesPerson() As String
    Public Property Quantity() As Integer
End Class